### PR TITLE
Cleanup matrix setup in miniapp_triangular_multiplication to enable complex types

### PR DESCRIPTION
**Pull request by gulivarese**

### CODE REVIEW

File: miniapp/miniapp_triangular_multiplication.cpp [hunk 1]
```diff
@@ -95,37 +97,57 @@ linear_system_t<T> sampleLeftTr(blas::Uplo uplo, blas::Op op, blas::Diag diag, T
...
-
-    const auto uplo = opts.uplo;
-    const auto op = opts.op;
-    const auto diag = opts.diag;
     const T alpha = 2.0;
 
-    double m = ah.size().rows();
-    double n = bh.size().cols();
-    auto add_mul = n * m * m / 2;
     const double total_ops = dlaf::total_ops<T>(add_mul, add_mul);
```
Comment (rasolca) [inline miniapp/miniapp_triangular_multiplication.cpp:151]: How can this compile now?

File: miniapp/miniapp_triangular_multiplication.cpp [hunk 2]
```diff
@@ -95,37 +97,57 @@ linear_system_t<T> sampleLeftTr(blas::Uplo uplo, blas::Op op, blas::Diag diag, T
...
 
     auto [in_op_a, out_b, in_b] = ::sampleLeftTr(uplo, op, diag, alpha, ah.size().rows());
```
Comment (rasolca) [inline miniapp/miniapp_triangular_multiplication.cpp:153]: Where do you still need it?

File: miniapp/miniapp_triangular_multiplication.cpp [hunk 3]
```diff
@@ -220,70 +236,4 @@ int main(int argc, char** argv) {
...
-      return ((kk - 1) * gamma + el_x(index)) / alpha;
-    else
-      return kk * gamma / alpha;
-  };
-
-  return {el_op_a, el_b, el_x};
-}
-
-}
+}
```
Comment (rasolca) [inline miniapp/miniapp_triangular_multiplication.cpp:239]: new line needed at the end of file.

**Comment by gulivarese** [comment]
cscs-ci run

File: miniapp/miniapp_triangular_multiplication.cpp [hunk 4]
```diff
@@ -169,9 +189,6 @@ struct triangularMultiplicationMiniapp {
                   << pika::get_os_thread_count() << " " << backend << std::endl;
       }
 
-      b.copyTargetToSource();
-
-      // (optional) run test
       if ((opts.do_check == dlaf::miniapp::CheckIterFreq::Last && run_index == (opts.nruns - 1)) ||
           opts.do_check == dlaf::miniapp::CheckIterFreq::All) {
         DLAF_UNIMPLEMENTED("Check");
```
Comment (msimberg) [inline miniapp/miniapp_triangular_multiplication.cpp:194]: Confused by the title of the PR: "Miniapp trmm check". Is this supposed to enable results checking for the miniapp or is that to be done in a separate PR? If separate PR, could you please update the title and description of the PR to say what the goal of this PR is?

**Comment by gulivarese** [comment]
cscs-ci run

File: miniapp/miniapp_triangular_multiplication.cpp [hunk 5]
```diff
@@ -221,69 +236,3 @@ int main(int argc, char** argv) {
...
-/// B_ij = (Sum_k op(A)_ik * X_kj) / alpha
-///      = (op(A)_ii * X_ij + (kk-1) * gamma) / alpha,
-/// where gamma = (i+1) / (j+2) * exp(I*(2*i+j)),
-///       kk = i+1 if op(a) is an lower triangular matrix, or
-///       kk = m-i if op(a) is an lower triangular matrix.
-/// Therefore
-/// B_ij = (X_ij + (kk-1) * gamma) / alpha, if diag == Unit
-/// B_ij = kk * gamma / alpha, otherwise.
-template <typename T>
-linear_system_t<T> sampleLeftTr(blas::Uplo uplo, blas::Op op, blas::Diag diag, T alpha, SizeType m) {
```
Comment (msimberg) [inline miniapp/miniapp_triangular_multiplication.cpp:249]: If this function is no longer needed, can we remove the forward declaration further up the file as well (including whatever typedefs etc. the declaration requires)?

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Review by albestro — COMMENTED** [review]
@rasolca should we wait for #1175 and drop all clang-format workarounds?

**Comment by rasolca** [comment]
cscs-ci run
